### PR TITLE
Phpunit8 compat

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -1,15 +1,20 @@
 <?xml version="1.0" encoding="UTF-8"?>
-
-<phpunit colors="true" bootstrap="vendor/autoload.php">
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/8.0/phpunit.xsd"
```
Comment (greg0ire) [inline phpunit.xml.dist]: ah let me fix this actually

File: phpunit.xml.dist [hunk 2]
```diff
@@ -1,15 +1,17 @@
 <?xml version="1.0" encoding="UTF-8"?>
-
-<phpunit colors="true" bootstrap="vendor/autoload.php">
-    <testsuites>
```
Comment (mwadon) [inline phpunit.xml.dist]: Are you sure that you don't need `<testsuites>` element? What do you think @Nyholm ?

Comment (greg0ire) [reply phpunit.xml.dist]: phpunit itself generates it like that. Try running `phpunit --generate`.

Also, see how the test suite is recognized, and how it is not when I make a typo:

```
phpunit --testsuite MatthiasSymfonyConfigTest
PHPUnit 7.5.1 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.14
Configuration: /home/greg/dev/SymfonyConfigTest/phpunit.xml.dist

.......................................                           39 / 39 (100%)

Time: 66 ms, Memory: 10.00MB

OK (39 tests, 65 assertions)

phpunit --testsuite MatthiasSymfonyConfigTes 
PHPUnit 7.5.1 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.14
Configuration: /home/greg/dev/SymfonyConfigTest/phpunit.xml.dist



Time: 52 ms, Memory: 10.00MB

No tests executed!

```

Comment (mwadon) [reply phpunit.xml.dist]: Didn't know 👍

Comment (greg0ire) [reply phpunit.xml.dist]: Me neither until yesterday ^^

**Comment by greg0ire** [comment]
Wait a bit before merging, I want to fully understand, there are weird things: https://github.com/sebastianbergmann/phpunit/commit/7bca1e7d27f79483b073d87d8ec4ee08f5adb68c

**Comment by greg0ire** [comment]
I remember generating this with phpunit 7.0.0, so that we are compatible with it, but maybe I should do it with the latest phpunit 7?

**Comment by greg0ire** [comment]
With phpunit 7.5.3:
```
<?xml version="1.0" encoding="UTF-8"?>
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/7.5/phpunit.xsd"
         bootstrap="vendor/autoload.php"
         forceCoversAnnotation="true"
         beStrictAboutCoversAnnotation="true"
         beStrictAboutOutputDuringTests="true"
         beStrictAboutTodoAnnotatedTests="true"
         verbose="true">
    <testsuites>
        <testsuite name="default">
            <directory suffix="Test.php">Tests</directory>
        </testsuite>
    </testsuites>

    <filter>
        <whitelist processUncoveredFilesFromWhitelist="true">
            <directory suffix=".php">.</directory>
        </whitelist>
    </filter>
</phpunit>
```

**Comment by greg0ire** [comment]
I restored the `testsuites` element but kept `7.0` as a version for the XSD schema. phpunit 7 behaves the same with and without a typo. Besides the schema at https://schema.phpunit.de/7.0/phpunit.xsd recognizes the `testsuites` element as valid.

**Comment by greg0ire** [comment]
I also restored `colors=true` because it's nicer.

**Comment by emodric** [comment]
@greg0ire Hm, are you sure this will work?

The following script:

```php
<?php

abstract class Foo 
{
    public function __construct()
    {   
    }   
}

abstract class Bar 
{
}

var_dump(is_callable('Foo', '__construct'));
var_dump(is_callable('Bar', '__construct'));
```

results in:

```
bool(true)
bool(true)
```

**Comment by greg0ire** [comment]
Yes, it works: https://3v4l.org/NG586q

In the code you tried, there is no `extends`, and you test this from outside the object. And this package requires php 7.

**Comment by emodric** [comment]
Right, that explains it :+1:

**Comment by greg0ire** [comment]
Can someone please merge?

**Comment by greg0ire** [comment]
Thank you! Can someone please tag a release with this?

**Comment by mwadon** [comment]
@greg0ire done: 4.0.1

**Comment by greg0ire** [comment]
Great! Thanks a lot!

**Pull request merged**